tracks/java/exercises/isogram/mentoring.md

```diff
@@ -0,0 +1,40 @@
+# Mentoring
+
+## Reasonable Solutions
+
+### Using Streams
+
+```java
+public class IsogramChecker {
+    public boolean isIsogram(String word) {
+
+        String noSpacesOrHyphens = word.replaceAll("(?:\\s|-)", "").toLowerCase();
+
+        return noSpacesOrHyphens.chars().distinct().count() == noSpacesOrHyphens.length();        
+    }
+}
+
+```
+
+### Using a Set
+
+```java
+class IsogramChecker {
+  public boolean isIsogram(String input) {
+    Set<Character> tmp = new HashSet<Character>();
+    for(char ch : input.toLowerCase().toCharArray()) {
+        if (Character.isLetter(ch) && !tmp.add(ch)) {
+            return false;
+        }
+    }
+    return true;
+  }
+}
+```
+
+## Talking points
+
+- Solutions that use the Collections Framework should opt for a Set interface to check for duplicates instead of Maps and Lists. Sets don’t allow duplicates, so they are the best fit for this check.
+- The `add()` method in the Set interface returns a boolean value indicating if the value is already present or not. This can be used to stop code execution earlier, instead of building the entire Set and later on checking its size against the original String.
```

Comment:
```suggestion
- The `add()` method in the Set interface returns a Boolean value indicating if the value is already present or not. This can be used to stop code execution earlier, instead of building the entire Set and later on checking its size against the original String.
```

Reply:
Since the return is a primitive boolean, I think we should keep the word in lowercase. `Boolean` would imply a wrapper object.

Reply:
Then should it be wrapped in a code block marking?  `boolean` because it is a primitive Boolean, as opposed to a wrapper named `Boolean` that is treated as a Boolean?

Reply:
Exactly, it is a primitive. There is a clear distinction in Java between `boolean` and `Boolean`. I'll add the code markings, thanks for the suggestion.